substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/elf/dwarf/DwarfLineSectionImpl.java

```diff
@@ -478,6 +479,32 @@ private int writeLineNumberTable(DebugContext context, ClassEntry classEntry, by
             }
             pos = writeCopyOp(context, buffer, pos);
 
+            /*
+             * On AArch64 gdb expects to see a line record at the start of the method and a second
+             * one at the end of the prologue marking the point where the method code begins for
+             * real. If we don't provide it then gdb will skip to the second line record when we
+             * place a breakpoint on the method.
+             *
+             * We can identify the end of the prologue for normal methods by noting where the stack
+             * frame height is first adjusted. This should normally be no more a few instructions in
+             * total.
+             */
+            if (isAArch64() && !primaryEntry.getFrameSizeInfos().isEmpty()) {
```

Comment:
If doing the same also for AMD64 does not have any negative side effects then maybe it's not worth to make this piece architecture specific.